pkg/controller/issueapi/issue.go

```diff
@@ -70,23 +74,31 @@ func (c *Controller) IssueMany(ctx context.Context, requests []*api.IssueCodeReq
 		results[i] = c.IssueCode(ctx, vCode, realm)
 	}
 
-	// Send SMS messages
-	var wg sync.WaitGroup
-	for i, result := range results {
-		if result.ErrorReturn != nil {
-			continue
-		}
+	defer c.recordStats(ctx, results)
 
-		wg.Add(1)
-		go func(request *api.IssueCodeRequest, r *IssueResult) {
-			defer wg.Done()
-			c.SendSMS(ctx, request, r, realm)
-		}(requests[i], result)
+	// Send SMS messages if there's an SMS provider.
+	smsProvider, err := c.smsProviderFor(ctx, realm)
```

Comment:
ah one sms provider here... why didn't I think of this.

Reply:
this is lgtm, but there is an unhappy unit test

Reply:
I'm now wondering if a pool of connections (rather than one or always making new ones) is what we really want

Reply:
Go's HTTP client re-uses connections, and Twilio is HTTP/2. Given we didn't see the system fall over with batch upload when creating entirely new connections for each request, I don't anticipate we need connection pooling for the size/number of requests. Connection pooling tends to introduce a bunch of complexity (see database layer with idle timeouts, max timeouts, etc) that seems like unnecessary overhead for this until we actually need it 😄